CONTRIBUTING.md

```diff
@@ -1,16 +1,225 @@
+* [Writing Documentation](#writing-documentation)
 
-3. Comments on a pull request should be added to the request itself, and *not* to the commit.
+If you have questions, please send an email to users@riot-os.org or
+devel@riot-os.org mailing list or chat on
+[#riot-os](http://webchat.freenode.net?channels=%23riot-os).
 
-4. Keep commits to the point, e.g., don't add whitespace/typo fixes to other code changes. If changes are layered, layer the patches.
+As a reminder, all contributors are expected to follow our
+[Code of Conduct](CODE_OF_CONDUCT.md).
 
-5. Describe the technical detail of the change(s) as specific as possible.
+## Feature Requests
+[feature-requests]: #feature-requests
 
-6. Use [Labels](https://github.com/RIOT-OS/RIOT/wiki/Labels) to help classify pull requests and issues.
+Before filling a new feature request, check into the
+[existing feature requests](https://github.com/RIOT-OS/RIOT/issues?q=is%3Aopen+is%3Aissue+label%3A%22Feature+Request%22)
+if there's one already opened on the same topic.
 
-7. Please adhere to our [code of conduct](https://github.com/RIOT-OS/RIOT/blob/master/CODE_OF_CONDUCT.md)
+To request new features or enhancements, just fill a new
+[feature request issue](https://github.com/RIOT-OS/RIOT/issues/new?template=feature_request.md).
+Describe your needs and use cases if any. Give detail on why this is important
+and/or necessary for RIOT.
+
+## Bug Reports
+[bug-reports]: #bug-reports
+
+While bugs are unfortunate, they're a reality in software. We can't fix what we
+don't know about, so please report liberally. If you're not sure if something
+is a bug or not, feel free to file a bug anyway.
+
+**If you believe reporting your bug publicly represents a security risk to
+RIOT users, please send an email describing the bug to security@riot-os.org**.
+
+Before reporting a bug, have a look at
+[opened bugs](https://github.com/RIOT-OS/RIOT/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3Abug+).
+Maybe someone has already reported your error.
+
+Once you have verified this, opening an issue is as easy as clicking on [this
+link](https://github.com/RIOT-OS/RIOT/issues/new?template=bug_report.md) and
+filling out the fields.
+
+Each bug report issue uses a template with 5 sections that are here to help
+maintainer understand your issue and eventually reproduce it:
+
+    #### Description
+
+    #### Steps to reproduce the issue
+
+    #### Expected results
+
+    #### Actual results
+
+    #### Versions
+
+To fill the `version` section, use the script provide in the RIOT git
+repository:
+```
+./dist/tools/ci/print_toolchain_versions.sh
+```
+
+In summary, try to put as much information as possible that you think could
+help maintainers or other developpers to fix the bug quickly.
+
+## Pull Requests
+[pull-requests]: #pull-requests
+
+Pull Requests (PR) are the primary mechanism used to change RIOT. GitHub itself
+has some [great documentation][about-pull-requests] on using the Pull Request
```

Comment:
As far as I know, generic link labels can be less useful for people using screen readers (This [Article on 'click here' links explains why](https://webaccess.berkeley.edu/ask-pecan/click-here)). Using "using the Pull Request Feature" as your link text instead of "great documentation" could remedy this